docs/CONTRIBUTING.md

```diff
@@ -12,22 +12,70 @@ request.
 
 It is important to note that there are **two types of PRs** for this repository:
 
-1. **New Userstyles:** Please see [userstyle-creation.md](userstyle-creation.md).
-2. **Other improvements:** These should follow the recommendations below, to help
-   improve the chances of a merged PR.
+1. **New Userstyles:** Please see
+   [userstyle-creation.md](./userstyle-creation.md).
+2. **Other improvements:** These should follow the recommendations below, to
+   help improve the chances of a merged PR.
+## Recommendations
 
-## General Recommendations
+### Root Variables
+
+Whilst writing an userstyle, its likely you have come across something like
+`--black: #000;` in the `:root` selector. These are called root variables, and
+they are used to store values that can be reused throughout the userstyle. In
+userstyles, its important that we prefer using these over theming individual
+elements, as it allows for a more consistent and maintainable userstyle.
+
+### Opinionated Changes
+
+When making changes to the userstyle, it is important to remember that different
+users would like to have different things. This is why we have a set of rules
+that we follow when making changes to the userstyle. This is why we have the
+linter to flag any changes that do not follow these rules. These rules include:
+
+- No font changes
+- No layout changes
+  - Padding, margin, and width changes
+  - Hiding elements
+  - Changes to the position of elements
```

Comment:
```suggestion
When writing or updating a userstyle, it is important to keep in mind that different users have different preferences. To avoid lengthy discussion over user interface aesthetics, we have a set of rules for what a userstyle may include; importantly, **changes to font or layout (padding, margin, hiding/moving elements) are prohibited.**
```